agent/job_runner.go

```diff
@@ -392,6 +394,36 @@ func (r *JobRunner) normalizeVerificationBehavior(behavior string) (string, erro
 	}
 }
 
+func filterEnv(env map[string]string, allowedVariables []string) (map[string]string, []string) {
+	// Copy all environment variables to the new map
+	// So it won't modify the original list
+	var environmentVariables = make(map[string]string, len(env))
+	for k := range env {
+		environmentVariables[k] = env[k]
+	}
+
+	var deniedVariables []string
```

Comment:
nit: keep it tight (var deniedVariables). no need for additional whitespaces as this parts of code a logically together.
```
// Filter out if the user has unlisted environment variables
// It adds environment variable to denied list and removes if it finds matching regex
var deniedVariables []string
for k := range environmentVariables {
		deniedVariables = append(deniedVariables, k)
		for _, v := range allowedVariables {
			if isMatch, _ := regexp.MatchString(v, k); isMatch {
				deniedVariables = deniedVariables[:len(deniedVariables)-1]
				break
			}
		}
	}
```